tests/testthat/test-exclusions.R

```diff
@@ -68,3 +73,22 @@ test_that("it gives the expected error message when there is mismatch between mu
     fixed = TRUE
   )
 })
+
+test_that("partial matching works for exclusions but warns if no linter found", {
+  read_settings(NULL)
+
+  expect_warning(
```

Comment:
is this really the way to do multiple warning tests in testthat? oh boy 😂

the withCallingHandlers approach is not particularly nicer... maybe we need a helper. i've definitely written `expect_warnings()` before.

otoh, it seems like an intentional design decision that testthat has limited matching exactly one warning at a time:

https://testthat.r-lib.org/reference/expect_error.html

Reply:
Filed https://github.com/r-lib/testthat/issues/1625 as a suggestion